autogpt_platform/frontend/src/lib/autogpt-server-api/baseClient.ts

```diff
@@ -195,6 +195,15 @@ export default class BaseAutoGPTServerAPI {
         response_data.detail,
         response,
       );
+
+      if (
+        response.status === 403 &&
+        response_data.detail === "Not authenticated" &&
+        window // Browser environment only: redirect to login page.
+      ) {
+        window.location.href = "/login";
+      }
+
```

Comment:
probably not the right way to do this... I'd do filtering on the frontend to only show those pages to logged in users but this is a workaround I suppose. Don't love this at all. Add a ticket to do this right

Reply:
As I described below. The complexity comes from the the feature toggle for auth only present on the server-side.

You won't be able to know whether you need an auth/user only when your request is failing. The reason the UX remains broken is also likely due to this.

There is another complexity: you can't do fetch redirect: follow, from server-side either. Since the API is running on different origin server than the application.

Losing state for logged out user I think it's much better than retaining the state, otherwise you might still be storing unwanted information.

Unless we are reworking the UX, I can't think of a better approach for this. I'm open if you have another alternative that can fix this issue, while not breaking a local user that doesn't enable server side auth (this is the default).